### ISSUE TITLE
[WIP] Transaction confirmation screen

### ISSUE THREAD

**Issue by maht0rz** [galacticcouncil/Basilisk-ui-old#44]
### Motivation
As a user, i want to be able to confirm a transaction before it is sent to the [Polkadot.js extension](https://polkadot.js.org/docs/extension) for signing.

### Links
![](https://i.imgur.com/SDrdRga.png)
[Flow diagram](https://www.figma.com/file/19w8uPggtCAbToCmFpVwZw/Untitled?node-id=0%3A1)

#### Account
|Attribute|Type|Example Value|Note|
|---|---|---|---|
|**isActive**|`boolean`|`true/false`|-|

#### Action
|Attribute|Type|Example Value|Note|
|---|---|---|---|
|**result**|`'cancelled' / 'failed' / 'successful'`|`'cancelled' / 'failed' / 'successful'`|-|

### Detailed description

- Provide a generic purpose confirmation screen, that parses out an extrinsic call before it is `signAndSend`-ed
- Parse out the palette, method and args information together with docs. Show it to the user in a manner concise with what they can see in the Polkadot.js extension
- Parse our encoded call data and show it to the user
    - You can find out [how to parse the call data from here](https://github.com/polkadot-js/apps/blob/master/packages/page-extrinsics/src/Submission.tsx#L45)
- Parse our encoded call hash and show it to the user
    - You can find out [how to parse the call hash from here](https://github.com/polkadot-js/apps/blob/master/packages/page-extrinsics/src/Submission.tsx#L45)
- Show payment info
    - Show the payment into in the desired fee payment asset
- Show an input for a tip (TODO: tip task)
- Show nonce
- TODO: show lifetime input/info (?)
- Show active account (signer for the transaction)
- In order to display the same information as the extension will, we need to forcefully update extension metadata each time the user tries to sign something via the confirmation screen (only after pressing the 'sign' button)
    - Older implementation (in the hydra claims ui) can be [found here](https://github.com/galacticcouncil/HydraDX-claims/blob/0b8b5c683ab251ef5c8b37a98c104973f91bd1c6/src/services/polkadotExtension.ts#L140) TODO: separate issue for metadata updates
- The transaction confirmation screen needs to preceed any extrinsics call, only on user confirmation the extrinsic should be sent to the extension for actual extension signing.
- In a case when the extension throws an error while signing, then:
    - If its an error that should be gracefully handled (e.g. natural cancel or extension closing), then the confirmation screen should be closed as well
    - If its an error that is not known (not gracefully handled), then the extension should display a screen saying something such as 'Something went wrong while signing'. A retry option should be offered at this time too.

#### Parsing of docs
```
const call = apiInstance.tx.xyk.buy(
  '0',
  '1',
  '1000000000000000000000000000000000000',
  '10000000',
  false
);

console.log(call.meta.docs.map(text => text.toString()).filter(text => text.length).join(' '));
```

**Comment by Cubo25** [comment]
Closing due to clean up.
